src/messages/MAuth.h

```diff
@@ -33,7 +33,7 @@ class MAuth : public PaxosServiceMessage {
 
   MAuth() : PaxosServiceMessage{CEPH_MSG_AUTH, 0}, protocol(0), monmap_epoch(0) { }
 private:
-  ~MAuth() override {}
+  ~MAuth() override final {}
```

Comment:
might want to s/override final/final/. as `final` specifier ensure that `~MAuth` overrides the dtor of its base class.

Reply:
Is this a point specific to `MAuth`? My understanding is that `final` does not imply `override`. We want `override` to indicate the parent's dtor must be virtual?

Reply:
no, it's not specific to `MAuth`. it applies to all "override final"s. according to N4659, §12.2

> A *virt-specifier-seq* shall contain at most one of each *virt-specifier*. A *virt-specifier-seq* shall appear only in the declaration of a virtual member function.

so if we use `final` as the virt-specifier-seq, there is no need to put "override". in other words, if we put `final` on a non virtual function, we will have a compiling error, like
```
test.cc:6:8: error: ‘void derived_t::func()’ marked ‘final’, but is not virtual
    6 |   void func() final {}
      |        ^~~~
```

i agree that that `final` does not imply `override`. but it practically implies "override".

Reply:
Thanks for explaining @tchaikov. I've made this change.